In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
# Permite imprimir plots directamente en el notebook
%matplotlib inline

In [5]:
# Opciones de visualizacion

pd.options.display.max_rows = 10
pd.options.display.float_format = '{:,.2f}'.format
pd.set_option('display.max_rows', 30)
plt.rcParams['figure.figsize'] = (10, 10)

In [6]:
# Lectura de datasets
salud_cols = ['DIRECTORIO', 'NPCFP1', 'NPCFP2', 'NPCFP7',
              'NPCFP13A', 'NPCFP13B', 'NPCFP13D', 'NPCFP13E', 'NPCFP14B']
vivienda_cols = ['DIRECTORIO', 'NVCBP8A', 'NVCBP8B',
                'NVCBP8D', 'NVCBP8G', 'NVCBP8H', 'NVCBP10', 'NVCBP11AA',
                'NVCBP11B', 'NVCBP11C', 'NVCBP11D', 'NVCBP12', 'NVCBP13',
                'NVCBP14A', 'NVCBP14B', 'NVCBP14I', 'NVCBP15D', 'NVCBP15E',
                'NVCBP15F']

salud = pd.read_csv('data/f_salud.csv', sep=';', low_memory=False)[salud_cols]
vivienda = pd.read_csv('data/b_vivienda_y_entorno.csv', sep=';', low_memory=False)[vivienda_cols]

In [7]:
salud.shape

(319952, 9)

In [109]:
salud.head()

DIRECTORIO  NPCFP1 NPCFP2 NPCFP7 NPCFP13A NPCFP13B NPCFP13D NPCFP13E  \
0      101000       1      1      2        1                              
1      101000       1      1      2        1        1        1            
2      101000       1      1      2        1                              
3      101001       1      2      1                                   1   
4      101001       1      2      2        1                              

   NPCFP14B  
0         2  
1         2  
2         2  
3         2  
4         2

In [8]:
vivienda.shape

(107218, 19)

In [110]:
vivienda.head()

DIRECTORIO  NVCBP8A  NVCBP8B  NVCBP8D  NVCBP8G  NVCBP8H  NVCBP10 NVCBP11AA  \
0      101000        2        2        2        2        2        2         2   
1      101001        1        1        2        2        1        2         3   
2      101002        2        2        2        2        2        1         3   
3      101003        1        2        2        1        1        1         3   
4      101004        2        2        2        2        2        2         6   

   NVCBP11B  NVCBP11C  NVCBP11D  NVCBP12  NVCBP13  NVCBP14A  NVCBP14B  \
0         1         1         1        1        4         2         2   
1         1         1         1        1        2         2         2   
2         1         1         1        1        4         2         2   
3         1         1         1        1        4         2         2   
4         1         1         1        1        1         2         2   

   NVCBP14I  NVCBP15D  NVCBP15E  NVCBP15F  
0         2         2         1         2  
1         2         2         2         2  
2         2         2         2         2  
3         1         1         2         2  
4         1         2         2         2

# PREPARACION DE LOS DATOS

1. Los nombres de las columnas para cada data set están codificados por lo que se dificulta su interpretación.
2. Datos nulos o vacíos.

In [111]:
salud_col_rename={
    "NPCFP1" : "SEGURIDAD_SOCIAL_SI",
    "NPCFP2" : "SEGURIDAD_SOCIAL_REGIMEN",
    "NPCFP7" : "PARTICIPA_PROMO_PREVE",
    "NPCFP13A" : "CONSULTA_MEDICINA_GENERAL",
    "NPCFP13B" : "CONSULTA_MEDICINA_ESPECIALIZADA",
    "NPCFP13D" : "CONSULTA_MEDICINA_ALTERNATIVA",
    "NPCFP13E" : "CONSULTA",
    "NPCFP14B" : "TARGET_ENFERMEDAD_RESPIRATORIA"
}

salud_encoded = salud.rename(columns=salud_col_rename, errors="raise")
salud_encoded.head()

DIRECTORIO  SEGURIDAD_SOCIAL_SI SEGURIDAD_SOCIAL_REGIMEN  \
0      101000                    1                        1   
1      101000                    1                        1   
2      101000                    1                        1   
3      101001                    1                        2   
4      101001                    1                        2   

  PARTICIPA_PROMO_PREVE CONSULTA_MEDICINA_GENERAL  \
0                     2                         1   
1                     2                         1   
2                     2                         1   
3                     1                             
4                     2                         1   

  CONSULTA_MEDICINA_ESPECIALIZADA CONSULTA_MEDICINA_ALTERNATIVA CONSULTA  \
0                                                                          
1                               1                             1            
2                                                                          
3                                                                      1   
4                                                                          

   TARGET_ENFERMEDAD_RESPIRATORIA  
0                               2  
1                               2  
2                               2  
3                               2  
4                               2

In [112]:
vivienda_col_rename={
    "NVCBP8A" : "VIV_HUMEDAD",
    "NVCBP8B" : "VIV_GOTERAS",
    "NVCBP8D" : "VIV_FALLO_TUBERIAS",
    "NVCBP8G" : "VIV_POCA_VENTILACION",
    "NVCBP8H" : "VIV_INUNDACION",
    "NVCBP10" : "VIV_TIPO",
    "NVCBP11AA" : "VIV_ESTRATO",
    "NVCBP11B" : "VIV_ACUEDUCTO",
    "NVCBP11C" : "VIV_ALCANTARILLADO",
    "NVCBP11D" : "VIV_RECOLECCION_BASURA",
    "NVCBP12" : "VIV_MATERIAL_PARED",
    "NVCBP13" : "VIV_MATERIAL_PISO",
    "NVCBP14A" : "VIV_CERCANIA_FABRICAS",
    "NVCBP14B" : "VIV_CERCANIA_BOTADEROS",
    "NVCBP14I" : "VIV_CERCANIA_CANO_AGUAS",
    "NVCBP15D" : "SECTOR_CONTAMINACION_AIRE",
    "NVCBP15E" : "SECTOR_MAL_OLOR",
    "NVCBP15F" : "SECTOR_MAL_MANEJO_BASURAS"
}

vivienda_encoded = vivienda.rename(columns=vivienda_col_rename, errors="raise")
vivienda_encoded.head()

DIRECTORIO  VIV_HUMEDAD  VIV_GOTERAS  VIV_FALLO_TUBERIAS  \
0      101000            2            2                   2   
1      101001            1            1                   2   
2      101002            2            2                   2   
3      101003            1            2                   2   
4      101004            2            2                   2   

   VIV_POCA_VENTILACION  VIV_INUNDACION  VIV_TIPO VIV_ESTRATO  VIV_ACUEDUCTO  \
0                     2               2         2           2              1   
1                     2               1         2           3              1   
2                     2               2         1           3              1   
3                     1               1         1           3              1   
4                     2               2         2           6              1   

   VIV_ALCANTARILLADO  VIV_RECOLECCION_BASURA  VIV_MATERIAL_PARED  \
0                   1                       1                   1   
1                   1                       1                   1   
2                   1                       1                   1   
3                   1                       1                   1   
4                   1                       1                   1   

   VIV_MATERIAL_PISO  VIV_CERCANIA_FABRICAS  VIV_CERCANIA_BOTADEROS  \
0                  4                      2                       2   
1                  2                      2                       2   
2                  4                      2                       2   
3                  4                      2                       2   
4                  1                      2                       2   

   VIV_CERCANIA_CANO_AGUAS  SECTOR_CONTAMINACION_AIRE  SECTOR_MAL_OLOR  \
0                        2                          2                1   
1                        2                          2                2   
2                        2                          2                2   
3                        1                          1                2   
4                        1                          2                2   

   SECTOR_MAL_MANEJO_BASURAS  
0                          2  
1                          2  
2                          2  
3                          2  
4                          2

## JOIN

Se unen los datos con base a la llave DIRECTORIO

In [3]:
pd.merge?

In [113]:
df = pd.merge(salud_encoded, vivienda_encoded, on='DIRECTORIO', how='inner').drop(columns=['DIRECTORIO'])

In [114]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 319952 entries, 0 to 319951
Data columns (total 26 columns):
SEGURIDAD_SOCIAL_SI                319952 non-null int64
SEGURIDAD_SOCIAL_REGIMEN           319952 non-null object
PARTICIPA_PROMO_PREVE              319952 non-null object
CONSULTA_MEDICINA_GENERAL          319952 non-null object
CONSULTA_MEDICINA_ESPECIALIZADA    319952 non-null object
CONSULTA_MEDICINA_ALTERNATIVA      319952 non-null object
CONSULTA                           319952 non-null object
TARGET_ENFERMEDAD_RESPIRATORIA     319952 non-null int64
VIV_HUMEDAD                        319952 non-null int64
VIV_GOTERAS                        319952 non-null int64
VIV_FALLO_TUBERIAS                 319952 non-null int64
VIV_POCA_VENTILACION               319952 non-null int64
VIV_INUNDACION                     319952 non-null int64
VIV_TIPO                           319952 non-null int64
VIV_ESTRATO                        319952 non-null object
VIV_ACUEDUCTO              

## Ocurrencias de Valores por Característica

In [115]:
def ver_ocurrencias():
    ocurrences_series = []
    
    for col in df:
        ocurrences_series.append(pd.Series(df[col].value_counts().values, index=df[col].value_counts().index.values, name=col))
    return pd.concat(ocurrences_series, axis=1)

df_ocurrencias = ver_ocurrencias()
df_ocurrencias

SEGURIDAD_SOCIAL_SI  SEGURIDAD_SOCIAL_REGIMEN  PARTICIPA_PROMO_PREVE  \
1           300,326.00                       nan                    nan   
2            17,066.00                       nan                    nan   
3                  nan                       nan                    nan   
4                  nan                       nan                    nan   
5                  nan                       nan                    nan   
6                  nan                       nan                    nan   
7                  nan                       nan                    nan   
8                  nan                       nan                    nan   
9             2,560.00                       nan                    nan   
                   nan                 19,626.00              21,769.00   
0                  nan                       nan                    nan   
1                  nan                215,468.00              30,057.00   
2                  nan                  7,908.00             268,126.00   
3                  nan                 74,807.00                    nan   
4                  nan                       nan                    nan   
5                  nan                       nan                    nan   
6                  nan                       nan                    nan   
8                  nan                       nan                    nan   
9                  nan                  2,143.00                    nan   

   CONSULTA_MEDICINA_GENERAL  CONSULTA_MEDICINA_ESPECIALIZADA  \
1                        nan                              nan   
2                        nan                              nan   
3                        nan                              nan   
4                        nan                              nan   
5                        nan                              nan   
6                        nan                              nan   
7                        nan                              nan   
8                        nan                              nan   
9                        nan                              nan   
                  104,479.00                       234,742.00   
0                        nan                              nan   
1                 215,473.00                        85,210.00   
2                        nan                              nan   
3                        nan                              nan   
4                        nan                              nan   
5                        nan                              nan   
6                        nan                              nan   
8                        nan                              nan   
9                        nan                              nan   

   CONSULTA_MEDICINA_ALTERNATIVA   CONSULTA  TARGET_ENFERMEDAD_RESPIRATORIA  \
1                            nan        nan                       10,388.00   
2                            nan        nan                      309,564.00   
3                            nan        nan                             nan   
4                            nan        nan                             nan   
5                            nan        nan                             nan   
6                            nan        nan                             nan   
7                            nan        nan                             nan   
8                            nan        nan                             nan   
9                            nan        nan                             nan   
                      313,700.00 227,633.00                             nan   
0                            nan        nan                             nan   
1                       6,252.00  92,319.00                             nan   
2                            nan        nan                             nan   
3                            nan        nan                             nan   
4               

In [116]:
df_ocurrencias.index

Index([1, 2, 3, 4, 5, 6, 7, 8, 9, ' ', '0', '1', '2', '3', '4', '5', '6', '8',
       '9'],
      dtype='object')

## Estandarización de tipo de dato

Se observa de la tabla de ocurrencias, que existen valores tanto numéricos como string. Para visualizar correctamente los valores, se aplicó una tranformación en el tipo de dato a cada columna del DataFrame.

In [117]:
for col in df:
    df[col] = df[col].apply(str)
    
df_ocurrencias_t = ver_ocurrencias().T.sort_index()

C:\Users\Andres\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  


In [118]:
df_ocurrencias_t

0          1          2  \
CONSULTA                        227,633.00    nan  92,319.00        nan   
CONSULTA_MEDICINA_ALTERNATIVA   313,700.00    nan   6,252.00        nan   
CONSULTA_MEDICINA_ESPECIALIZADA 234,742.00    nan  85,210.00        nan   
CONSULTA_MEDICINA_GENERAL       104,479.00    nan 215,473.00        nan   
PARTICIPA_PROMO_PREVE            21,769.00    nan  30,057.00 268,126.00   
SECTOR_CONTAMINACION_AIRE              nan    nan  94,855.00 225,097.00   
SECTOR_MAL_MANEJO_BASURAS              nan    nan  53,611.00 266,341.00   
SECTOR_MAL_OLOR                        nan    nan  97,451.00 222,501.00   
SEGURIDAD_SOCIAL_REGIMEN         19,626.00    nan 215,468.00   7,908.00   
SEGURIDAD_SOCIAL_SI                    nan    nan 300,326.00  17,066.00   
TARGET_ENFERMEDAD_RESPIRATORIA         nan    nan  10,388.00 309,564.00   
VIV_ACUEDUCTO                          nan    nan 316,566.00   3,386.00   
VIV_ALCANTARILLADO                     nan    nan 311,012.00   8,940.00   
VIV_CERCANIA_BOTADEROS                 nan    nan  26,112.00 293,840.00   
VIV_CERCANIA_CANO_AGUAS                nan    nan  39,213.00 280,739.00   
VIV_CERCANIA_FABRICAS                  nan    nan  42,521.00 277,431.00   
VIV_ESTRATO                         475.00 398.00  33,094.00 142,540.00   
VIV_FALLO_TUBERIAS                     nan    nan  18,004.00 296,559.00   
VIV_GOTERAS                            nan    nan  50,551.00 265,025.00   
VIV_HUMEDAD                            nan    nan  78,817.00 237,030.00   
VIV_INUNDACION                         nan    nan  13,661.00 301,197.00   
VIV_MATERIAL_PARED                     nan    nan 313,420.00   2,005.00   
VIV_MATERIAL_PISO                      nan    nan   3,749.00  13,754.00   
VIV_POCA_VENTILACION                   nan    nan  16,771.00 298,440.00   
VIV_RECOLECCION_BASURA                 nan    nan 314,718.00   5,234.00   
VIV_TIPO                               nan    nan 142,354.00 173,314.00   

                                         3          4        5         6  \
CONSULTA                               nan        nan      nan       nan   
CONSULTA_MEDICINA_ALTERNATIVA          nan        nan      nan       nan   
CONSULTA_MEDICINA_ESPECIALIZADA        nan        nan      nan       nan   
CONSULTA_MEDICINA_GENERAL              nan        nan      nan       nan   
PARTICIPA_PROMO_PREVE                  nan        nan      nan       nan   
SECTOR_CONTAMINACION_AIRE              nan        nan      nan       nan   
SECTOR_MAL_MANEJO_BASURAS              nan        nan      nan       nan   
SECTOR_MAL_OLOR                        nan        nan      nan       nan   
SEGURIDAD_SOCIAL_REGIMEN         74,807.00        nan      nan       nan   
SEGURIDAD_SOCIAL_SI                    nan        nan      nan       nan   
TARGET_ENFERMEDAD_RESPIRATORIA         nan        nan      nan       nan   
VIV_ACUEDUCTO                          nan        nan      nan       nan   
VIV_ALCANTARILLADO                     nan        nan      nan       nan   
VIV_CERCANIA_BOTADEROS                 nan        nan      nan       nan   
VIV_CERCANIA_CANO_AGUAS                nan        nan      nan       nan   
VIV_CERCANIA_FABRICAS                  nan        nan      nan       nan   
VIV_ESTRATO                     103,471.00  30,846.00 6,221.00  2,852.00   
VIV_FALLO_TUBERIAS                     nan        nan      nan       nan   
VIV_GOTERAS                            nan        nan      nan       nan   
VIV_HUMEDAD                            nan        nan      nan       nan   
VIV_INUNDACION                         nan        nan      nan       nan   
VIV_MATERIAL_PARED                  746.00     432.00 1,288.00  1,948.00   
VIV_MATERIAL_PISO                 2,616.00 263,968.00 6,009.00 29,067.00   
VIV_POCA_VENTILACION                   nan        nan      nan       nan   
VIV_RECOLECCION_BASURA                 nan        nan      nan       nan   
VIV_TIPO                          4,258.00    

Se comprueba que la suma de las categorías para cada feature, es igual al número de observaciones y no se cometió ningún error por ejemplo duplicando alguna columna.

In [119]:
df_ocurrencias_t.sum(axis=1)

CONSULTA                          319,952.00
CONSULTA_MEDICINA_ALTERNATIVA     319,952.00
CONSULTA_MEDICINA_ESPECIALIZADA   319,952.00
CONSULTA_MEDICINA_GENERAL         319,952.00
PARTICIPA_PROMO_PREVE             319,952.00
SECTOR_CONTAMINACION_AIRE         319,952.00
SECTOR_MAL_MANEJO_BASURAS         319,952.00
SECTOR_MAL_OLOR                   319,952.00
SEGURIDAD_SOCIAL_REGIMEN          319,952.00
SEGURIDAD_SOCIAL_SI               319,952.00
TARGET_ENFERMEDAD_RESPIRATORIA    319,952.00
VIV_ACUEDUCTO                     319,952.00
VIV_ALCANTARILLADO                319,952.00
VIV_CERCANIA_BOTADEROS            319,952.00
VIV_CERCANIA_CANO_AGUAS           319,952.00
VIV_CERCANIA_FABRICAS             319,952.00
VIV_ESTRATO                       319,952.00
VIV_FALLO_TUBERIAS                319,952.00
VIV_GOTERAS                       319,952.00
VIV_HUMEDAD                       319,952.00
VIV_INUNDACION                    319,952.00
VIV_MATERIAL_PARED                319,952.00
VIV_MATERI

## Valores no numéricos

Se observa del conjunto de datos que algunas columnas tienen el valor ' ', en particular las columnas:

In [120]:
df_ocurrencias_t[df_ocurrencias_t[' '].notna()]

0          1          2  \
CONSULTA                        227,633.00    nan  92,319.00        nan   
CONSULTA_MEDICINA_ALTERNATIVA   313,700.00    nan   6,252.00        nan   
CONSULTA_MEDICINA_ESPECIALIZADA 234,742.00    nan  85,210.00        nan   
CONSULTA_MEDICINA_GENERAL       104,479.00    nan 215,473.00        nan   
PARTICIPA_PROMO_PREVE            21,769.00    nan  30,057.00 268,126.00   
SEGURIDAD_SOCIAL_REGIMEN         19,626.00    nan 215,468.00   7,908.00   
VIV_ESTRATO                         475.00 398.00  33,094.00 142,540.00   

                                         3         4        5        6   7  \
CONSULTA                               nan       nan      nan      nan nan   
CONSULTA_MEDICINA_ALTERNATIVA          nan       nan      nan      nan nan   
CONSULTA_MEDICINA_ESPECIALIZADA        nan       nan      nan      nan nan   
CONSULTA_MEDICINA_GENERAL              nan       nan      nan      nan nan   
PARTICIPA_PROMO_PREVE                  nan       nan      nan      nan nan   
SEGURIDAD_SOCIAL_REGIMEN         74,807.00       nan      nan      nan nan   
VIV_ESTRATO                     103,471.00 30,846.00 6,221.00 2,852.00 nan   

                                    8        9  
CONSULTA                          nan      nan  
CONSULTA_MEDICINA_ALTERNATIVA     nan      nan  
CONSULTA_MEDICINA_ESPECIALIZADA   nan      nan  
CONSULTA_MEDICINA_GENERAL         nan      nan  
PARTICIPA_PROMO_PREVE             nan      nan  
SEGURIDAD_SOCIAL_REGIMEN          nan 2,143.00  
VIV_ESTRATO                     19.00    36.00

### Consultas

De acuerdo a la descripción de los datos, el valor 1 significa 'SI'. Se codificará este valor como 1 y ' ' como 0

In [121]:
df['CONSULTA'] = df['CONSULTA'].replace({
    ' ' : 'NO',
    '1' : 'SI'
})
df['CONSULTA_MEDICINA_GENERAL'] = df['CONSULTA_MEDICINA_GENERAL'].replace({
    ' ' : 'NO',
    '1' : 'SI'
})
df['CONSULTA_MEDICINA_ESPECIALIZADA'] = df['CONSULTA_MEDICINA_ESPECIALIZADA'].replace({
    ' ' : 'NO',
    '1' : 'SI'
})
df['CONSULTA_MEDICINA_ALTERNATIVA'] = df['CONSULTA_MEDICINA_ALTERNATIVA'].replace({
    ' ' : 'NO',
    '1' : 'SI'
})

### NPCFP7 (PARTICIPA_PROMO_PREVE)

De acuerdo a la descripción de datos: '1' significa SI y '2' significa NO. Sin embargo, en los datos se encuentran 21769 instancias con el valor vacío ' '.

In [122]:
df = df[df['PARTICIPA_PROMO_PREVE'] != ' ']
df['PARTICIPA_PROMO_PREVE'] = df['PARTICIPA_PROMO_PREVE'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

### NVCBP15D (SECTOR_CONTAMINACION_AIRE), NVCBP15E (SECTOR_MAL_MANEJO_BASURAS) y NVCBP15F (SECTOR_MAL_OLOR)

Para estas variables, '1' significa 'SI' y '2' 'NO'.

In [123]:
df['SECTOR_CONTAMINACION_AIRE'] = df['SECTOR_CONTAMINACION_AIRE'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['SECTOR_MAL_MANEJO_BASURAS'] = df['SECTOR_MAL_MANEJO_BASURAS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['SECTOR_MAL_OLOR'] = df['SECTOR_MAL_OLOR'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

### NPCFP1 (SEGURIDAD_SOCIAL_SI)

Vemos que todas las observaciones tienen el mismo valor, por lo que no aporta información

In [124]:
del df['SEGURIDAD_SOCIAL_SI']

### VIV_ACUEDUCTO, VIV_ALCANTARILLADO, VIV_RECOLECCION_BASURA, VIV_CERCANIA_FABRICAS, VIV_CERCANIA_BOTADEROS, VIV_CERCANIA_CANO_AGUAS

'1' significa 'SI', '2' significa 'NO'

In [125]:
df['VIV_ACUEDUCTO'] = df['VIV_ACUEDUCTO'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_ALCANTARILLADO'] = df['VIV_ALCANTARILLADO'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_RECOLECCION_BASURA'] = df['VIV_RECOLECCION_BASURA'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_CERCANIA_FABRICAS'] = df['VIV_CERCANIA_FABRICAS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_CERCANIA_BOTADEROS'] = df['VIV_CERCANIA_BOTADEROS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_CERCANIA_CANO_AGUAS'] = df['VIV_CERCANIA_CANO_AGUAS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

### VIV_FALLO_TUBERIAS, VIV_GOTERAS, VIV_HUMEDAD, VIV_INUNDACION, VIV_POCA_VENTILACION

'1' significa 'SI', '2' significa 'NO' y '9' No sabe no responde.

In [126]:
df['VIV_FALLO_TUBERIAS'] = df['VIV_FALLO_TUBERIAS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_GOTERAS'] = df['VIV_GOTERAS'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_HUMEDAD'] = df['VIV_HUMEDAD'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_INUNDACION'] = df['VIV_INUNDACION'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df['VIV_POCA_VENTILACION'] = df['VIV_POCA_VENTILACION'].replace({
    '2' : 'NO',
    '1' : 'SI'
})

df = df[df['VIV_FALLO_TUBERIAS'] != '9']
df = df[df['VIV_GOTERAS'] != '9']
df = df[df['VIV_HUMEDAD'] != '9']
df = df[df['VIV_INUNDACION'] != '9']
df = df[df['VIV_POCA_VENTILACION'] != '9']

df_ocurrencias_t = ver_ocurrencias().T.sort_index()
df_ocurrencias_t

C:\Users\Andres\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  


0          1          2         3  \
CONSULTA                           nan    nan        nan        nan       nan   
CONSULTA_MEDICINA_ALTERNATIVA      nan    nan        nan        nan       nan   
CONSULTA_MEDICINA_ESPECIALIZADA    nan    nan        nan        nan       nan   
CONSULTA_MEDICINA_GENERAL          nan    nan        nan        nan       nan   
PARTICIPA_PROMO_PREVE              nan    nan        nan        nan       nan   
SECTOR_CONTAMINACION_AIRE          nan    nan        nan        nan       nan   
SECTOR_MAL_MANEJO_BASURAS          nan    nan        nan        nan       nan   
SECTOR_MAL_OLOR                    nan    nan        nan        nan       nan   
SEGURIDAD_SOCIAL_REGIMEN           nan    nan 211,700.00   7,762.00 72,748.00   
TARGET_ENFERMEDAD_RESPIRATORIA     nan    nan   9,694.00 282,516.00       nan   
VIV_ACUEDUCTO                      nan    nan        nan        nan       nan   
VIV_ALCANTARILLADO                 nan    nan        nan        nan       nan   
VIV_CERCANIA_BOTADEROS             nan    nan        nan        nan       nan   
VIV_CERCANIA_CANO_AGUAS            nan    nan        nan        nan       nan   
VIV_CERCANIA_FABRICAS              nan    nan        nan        nan       nan   
VIV_ESTRATO                     370.00 347.00  29,483.00 129,547.00 94,806.00   
VIV_FALLO_TUBERIAS                 nan    nan        nan        nan       nan   
VIV_GOTERAS                        nan    nan        nan        nan       nan   
VIV_HUMEDAD                        nan    nan        nan        nan       nan   
VIV_INUNDACION                     nan    nan        nan        nan       nan   
VIV_MATERIAL_PARED                 nan    nan 286,335.00   1,835.00    688.00   
VIV_MATERIAL_PISO                  nan    nan   3,211.00  12,750.00  2,378.00   
VIV_POCA_VENTILACION               nan    nan        nan        nan       nan   
VIV_RECOLECCION_BASURA             nan    nan        nan        nan       nan   
VIV_TIPO                           nan    nan 129,868.00 158,767.00  3,555.00   

                                         4        5         6      7     8  \
CONSULTA                               nan      nan       nan    nan   nan   
CONSULTA_MEDICINA_ALTERNATIVA          nan      nan       nan    nan   nan   
CONSULTA_MEDICINA_ESPECIALIZADA        nan      nan       nan    nan   nan   
CONSULTA_MEDICINA_GENERAL              nan      nan       nan    nan   nan   
PARTICIPA_PROMO_PREVE                  nan      nan       nan    nan   nan   
SECTOR_CONTAMINACION_AIRE              nan      nan       nan    nan   nan   
SECTOR_MAL_MANEJO_BASURAS              nan      nan       nan    nan   nan   
SECTOR_MAL_OLOR                        nan      nan       nan    nan   nan   
SEGURIDAD_SOCIAL_REGIMEN               nan      nan       nan    nan   nan   
TARGET_ENFERMEDAD_RESPIRATORIA         nan      nan       nan    nan   nan   
VIV_ACUEDUCTO                          nan      nan       nan    nan   nan   
VIV_ALCANTARILLADO                     nan      nan       nan    nan   nan   
VIV_CERCANIA_BOTADEROS                 nan      nan       nan    nan   nan   
VIV_CERCANIA_CANO_AGUAS                nan      nan       nan    nan   nan   
VIV_CERCANIA_FABRICAS                  nan      nan       nan    nan   nan   
VIV_ESTRATO                      29,039.00 5,929.00  2,643.00    nan 13.00   
VIV_FALLO_TUBERIAS                     nan      nan       nan    nan   nan   
VIV_GOTERAS                            nan      nan       nan    nan   nan   
VIV_HUMEDAD                            nan      nan       nan    nan   nan   
VIV_INUNDACION                         nan      nan       nan    nan   nan   
VIV_MATERIAL_PARED                  391.00 1,118.00  1,758.00  17.00 59.00   
VIV_MATERIAL_PISO               241,690.00 5,490.00 26,071.00 620.00   nan   
VIV_POCA_VENTILACION                   nan      nan       nan    nan   nan   
VIV_RECOLECCION_BASURA                 nan      nan       nan    

### SEGURIDAD_SOCIAL_REGIMEN

1 contributivo
2 especial
3 subsidiado

In [127]:
df['SEGURIDAD_SOCIAL_REGIMEN'] = df['SEGURIDAD_SOCIAL_REGIMEN'].replace({
    '1' : 'contributivo',
    '2' : 'especial',
    '3' : 'subsidiado'
})

### Target (Enfermedad respiratoria)

1 SI, 2 NO

In [128]:
df['TARGET_ENFERMEDAD_RESPIRATORIA'] = df['TARGET_ENFERMEDAD_RESPIRATORIA'].replace({
    '1' : 'SI',
    '2' : 'NO',
})

### VIV_ESTRATO

0 Recibos sin estrato o el servicio es pirata, 1 Estrato 1, 2 Estrato 2, 3 Estrato 3, 4 Estrato 4, 5 Estrato 5, 6 Estrato
6, 8 Planta eléctrica, 9 No conoce el estrato o no cuenta con recibo de pago.

Esta es una variable ordinal categórica. Se codificará con un número entero (de acuerdo al estrato excepto que plantas eléctricas tomará el valor 7), y se descartará el valor 9.

In [129]:
df['VIV_ESTRATO'] = df['VIV_ESTRATO'].replace({
    '0' : 0,
    '1' : 1,
    '2' : 2,
    '3' : 3,
    '4' : 4,
    '5' : 5,
    '6' : 6,
    '8' : 7
})

df = df[df['VIV_ESTRATO'] != '9']

### VIV_MATERIAL_PARED

1 'rustico', 2 'tapia', 3 'bahareque_revocado', 4 'bahareque_sin_revocar', 5 'madera', 6 'prefabricado', 7 'natural', 8 'industrial', 9 'sin_paredes'

In [130]:
df['VIV_MATERIAL_PARED'] = df['VIV_MATERIAL_PARED'].replace({
    '1' : 'rustico',
    '2' : 'tapia',
    '3' : 'bahareque_revocado',
    '4' : 'bahareque_sin_revocar',
    '5' : 'madera',
    '6' : 'prefabricado',
    '7' : 'natural',
    '8' : 'industrial',
    '9' : 'sin_paredes'
})

### VIV_MATERIAL_PISO

1 'tapete', 2 'madera_pulida_laca', 3 'marmol', 4 'baldosa', 5 'madera_burda', 6 'cemento', 7 'tierra'

In [131]:
df['VIV_MATERIAL_PISO'] = df['VIV_MATERIAL_PISO'].replace({
    '1' : 'tapete',
    '2' : 'madera_pulida_laca',
    '3' : 'marmol',
    '4' : 'baldosa',
    '5' : 'madera_burda',
    '6' : 'cemento',
    '7' : 'tierra'
})

### VIV_TIPO

1 'casa', 2 'apartamento', 3 'cuarto', 4 'otro'

In [132]:
df['VIV_TIPO'] = df['VIV_TIPO'].replace({
    '1' : 'casa',
    '2' : 'apartamento',
    '3' : 'cuarto',
    '4' : 'otro'
})

In [133]:
df

SEGURIDAD_SOCIAL_REGIMEN PARTICIPA_PROMO_PREVE  \
0                  contributivo                    NO   
1                  contributivo                    NO   
2                  contributivo                    NO   
3                      especial                    SI   
4                      especial                    NO   
...                         ...                   ...   
319947             contributivo                    NO   
319948               subsidiado                    NO   
319949               subsidiado                    NO   
319950               subsidiado                    NO   
319951               subsidiado                    NO   

       CONSULTA_MEDICINA_GENERAL CONSULTA_MEDICINA_ESPECIALIZADA  \
0                             SI                              NO   
1                             SI                              SI   
2                             SI                              NO   
3                             NO                              NO   
4                             SI                              NO   
...                          ...                             ...   
319947                        SI                              NO   
319948                        NO                              NO   
319949                        NO                              NO   
319950                        NO                              NO   
319951                        NO                              NO   

       CONSULTA_MEDICINA_ALTERNATIVA CONSULTA TARGET_ENFERMEDAD_RESPIRATORIA  \
0                                 NO       NO                             NO   
1                                 SI       NO                             NO   
2                                 NO       NO                             NO   
3                                 NO       SI                             NO   
4                                 NO       NO                             NO   
...                              ...      ...                            ...   
319947                            NO       NO                             NO   
319948                            NO       SI                             NO   
319949                            NO       SI                             NO   
319950                            NO       SI                             NO   
319951                            NO       SI                             NO   

       VIV_HUMEDAD VIV_GOTERAS VIV_FALLO_TUBERIAS  ... VIV_ALCANTARILLADO  \
0               NO          NO                 NO  ...                 SI   
1               NO          NO                 NO  ...                 SI   
2               NO          NO                 NO  ...                 SI   
3               SI          SI                 NO  ...                 SI   
4               SI          SI                 NO  ...                 SI   
...            ...         ...                ...  ...                ...   
319947          NO          NO                 NO  ...                 NO   
319948          NO          NO                 NO  ...                 NO   
319949          NO          NO                 NO  ...                 NO   
319950          NO          NO                 NO  ...                 NO   
319951          NO          NO                 NO  ...                 NO   

       VIV_RECOLECCION_BASURA VIV_MATERIAL_PARED   VIV_MATERIAL_PISO  \
0                          SI            rustico             baldosa   
1                          SI            rustico             baldosa   
2                          SI            rustico             baldosa   
3                          SI            rustico  madera_pulida_laca   
4                          SI            rustico  madera_pulida_laca   
...                       ...                ...                 ...   
319947                     SI            rustico             baldosa   
319948                     NO            ru

In [134]:
ver_ocurrencias().T

C:\Users\Andres\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  


NO         SI  apartamento       casa  \
SEGURIDAD_SOCIAL_REGIMEN               nan        nan          nan        nan   
PARTICIPA_PROMO_PREVE           262,755.00  29,422.00          nan        nan   
CONSULTA_MEDICINA_GENERAL        89,252.00 202,925.00          nan        nan   
CONSULTA_MEDICINA_ESPECIALIZADA 210,472.00  81,705.00          nan        nan   
CONSULTA_MEDICINA_ALTERNATIVA   286,295.00   5,882.00          nan        nan   
CONSULTA                        214,090.00  78,087.00          nan        nan   
TARGET_ENFERMEDAD_RESPIRATORIA  282,483.00   9,694.00          nan        nan   
VIV_HUMEDAD                     220,379.00  71,798.00          nan        nan   
VIV_GOTERAS                     246,301.00  45,876.00          nan        nan   
VIV_FALLO_TUBERIAS              275,879.00  16,298.00          nan        nan   
VIV_POCA_VENTILACION            277,213.00  14,964.00          nan        nan   
VIV_INUNDACION                  279,698.00  12,479.00          nan        nan   
VIV_TIPO                               nan        nan   158,761.00 129,841.00   
VIV_ESTRATO                            nan        nan          nan        nan   
VIV_ACUEDUCTO                     2,894.00 289,283.00          nan        nan   
VIV_ALCANTARILLADO                7,733.00 284,444.00          nan        nan   
VIV_RECOLECCION_BASURA            4,446.00 287,731.00          nan        nan   
VIV_MATERIAL_PARED                     nan        nan          nan        nan   
VIV_MATERIAL_PISO                      nan        nan          nan        nan   
VIV_CERCANIA_FABRICAS           253,233.00  38,944.00          nan        nan   
VIV_CERCANIA_BOTADEROS          268,772.00  23,405.00          nan        nan   
VIV_CERCANIA_CANO_AGUAS         256,409.00  35,768.00          nan        nan   
SECTOR_CONTAMINACION_AIRE       205,640.00  86,537.00          nan        nan   
SECTOR_MAL_OLOR                 203,130.00  89,047.00          nan        nan   
SECTOR_MAL_MANEJO_BASURAS       243,494.00  48,683.00          nan        nan   

                                 contributivo   cuarto  especial  otro  \
SEGURIDAD_SOCIAL_REGIMEN           211,682.00      nan  7,762.00   nan   
PARTICIPA_PROMO_PREVE                     nan      nan       nan   nan   
CONSULTA_MEDICINA_GENERAL                 nan      nan       nan   nan   
CONSULTA_MEDICINA_ESPECIALIZADA           nan      nan       nan   nan   
CONSULTA_MEDICINA_ALTERNATIVA             nan      nan       nan   nan   
CONSULTA                                  nan      nan       nan   nan   
TARGET_ENFERMEDAD_RESPIRATORIA            nan      nan       nan   nan   
VIV_HUMEDAD                               nan      nan       nan   nan   
VIV_GOTERAS                               nan      nan       nan   nan   
VIV_FALLO_TUBERIAS                        nan      nan       nan   nan   
VIV_POCA_VENTILACION                      nan      nan       nan   nan   
VIV_INUNDACION                            nan      nan       nan   nan   
VIV_TIPO                                  nan 3,555.00       nan 20.00   
VIV_ESTRATO                               nan      nan       nan   nan   
VIV_ACUEDUCTO                             nan      nan       nan   nan   
VIV_ALCANTARILLADO                        nan      nan       nan   nan   
VIV_RECOLECCION_BASURA                    nan      nan       nan   nan   
VIV_MATERIAL_PARED                        nan      nan       nan   nan   
VIV_MATERIAL_PISO                         nan      nan       nan   nan   
VIV_CERCANIA_FABRICAS                     nan      nan       nan   nan   
VIV_CERCANIA_BOTADEROS                    nan      nan       nan   nan   
VIV_CERCANIA_CANO_AGUAS                   nan      nan       nan   nan   
SECTOR_CONTAMINACION_AIRE                 nan      nan       nan   nan   
SECTOR_MAL_OLOR                           nan      nan       nan   nan   
SECTOR_MAL_MANEJO_BASURAS                 nan      nan       nan   nan   

        

In [135]:
df.to_csv('data/data.limpia.csv', index=False)

In [137]:
df.VIV_ESTRATO.value_counts()

2    129547
3     94806
1     29483
4     29039
5      5929
6      2643
        370
0       347
7        13
Name: VIV_ESTRATO, dtype: int64